File: 백준/Silver/10845. 큐/큐.py
import sys
from collections import deque
r = sys.stdin.readline
sys.setrecursionlimit(10**6)

def solution():
    n = int(r().rstrip())
    q = deque()
    for _ in range(n):
        c = r().split()
        if c[0] == 'push':
            q.append(c[1])
        elif c[0] == 'pop':
            if q: print(q.popleft())
            else: print(-1)
        elif c[0] == 'size':
            print(len(q))
        elif c[0] == 'empty':
            if q: print(0)
            else: print(1)
        elif c[0] == 'front':
            if q: print(q[0])
            else: print(-1)
        else:
            if q: print(q[-1])
            else: print(-1)
    
solution()

File: 백준/Silver/10845. 큐/큐.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        
        int n = Integer.parseInt(br.readLine());
        Deque<Integer> q = new LinkedList<>();
        
        for(int i=0; i<n; i++) {
            String[] cmd = br.readLine().split(" ");
            
            switch (cmd[0]) {
                case "push":
                    q.addLast(Integer.parseInt(cmd[1]));
                    break;
                case "pop":
                    if(!q.isEmpty()) sb.append(q.removeFirst()).append('\n');
                    else sb.append(-1).append('\n');
                    break;
                case "size":
                    sb.append(q.size()).append('\n');
                    break;
                case "empty":
                    if(q.isEmpty()) sb.append(1).append('\n');
                    else sb.append(0).append('\n');
                    break;
                case "front":
                    if(!q.isEmpty()) sb.append(q.getFirst()).append('\n');
                    else sb.append(-1).append('\n');
                    break;
                case "back":
                    if(!q.isEmpty()) sb.append(q.getLast()).append('\n');
                    else sb.append(-1).append('\n');
                    break;
                default: break;
            }
        }
        System.out.println(sb.toString());
    }
}